tolstoy/src/metadata.rs

```diff
@@ -41,6 +46,14 @@ impl SyncMetadataClient {
         }
         Ok(())
     }
+
+    pub fn get_partitions(tx: &rusqlite::Transaction) -> Result<PartitionMap> {
```

Comment:
We have https://github.com/mozilla/mentat/blob/5fe4f12d32379415e36eb80877cbeec6fd443ffa/db/src/db.rs#L443 just waiting to be generalized; let's use it.